command/login.go

```diff
@@ -321,8 +321,8 @@ func (c *LoginCommand) Run(args []string) int {
 				"Vault.") + "\n")
 	}
 
-	if c.flagNoPrint {
-		return 0
+	if !c.flagPrintToken {
+		secret.Auth.ClientToken = "REDACTED"
```

Comment:
Is the `secret.Auth.ClientToken` used for subsequent requests? Perhaps a cleaner way to redact it would be to simply not call `OutputSecret` unless `c.flagPrintToken` is specified.